HPCC-16796 Fix configurator coredump

xwang2713 opened the pull request:
Configurator used to be built separately before docs build. Now it Is included in HPCC-Platform/configuration.
This pull request (JIRA HPCC-16796) includes:
1.  Fix two core dumps
1.  Enable configurator and docs build integration
1.  Change system/CMakeLists.txt to allow jlib only build for a component like configurator which depends on it.

The more detail information about configurator's fix will be documented in JIRA HPCC-16796

Test build: http://10.240.32.243/view/custom/job/Configurator-JIRA-16796/CE-DOCS=ubuntu-14.04-amd64-1/ws/build/Release/docs/

@g-pan please review generated docs
@Michael-Gardner please review the code

Signed-off-by: Xiaoming Wang <xiaoming.wang@lexisnexi.com>

ghalliday committed:
HPCC-16805 Start refactoring field and type formats

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
HPCC-16811 Provide functions to extract values from a typed field

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

shamser committed:
HPCC-15926 Add BEST attribute to DEDUP

Signed-off-by: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>

ghalliday committed:
HPCC-16244 Implement delayed heaplet link counting

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
HPCC-16244 Reimplement and optimize row compaction

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
HPCC-16517 Continue refactoring the roxie stats

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
HPCC-16517 Add some stats for some additional row managers

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
HPCC-16517 Track number of allocations for variable size rows

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
HPCC-16517 Changes following review

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

GordonSmith committed:
HPCC-16804 ECL Watch Flat Theme

Switch to new Dojo flat theme.

Fixes HPCC-16804

Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

jakesmith committed:
HPCC-16637 Save old environment when update detected.

Keep a copy of the old environment, to help diagnosing when
issues/changes are related to environment reconfiguration.

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>
Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

richardkchapman committed:
HPCC-16428 Coverity: resource leak

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Split off candidate-6.4.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.4.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-16603 Extract Java Doc comments from attributes.

And report via eclcc -M.

First stab to see if on the right path.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

Commit:
HPCC-14734 UI items login/logout in ECL Watch

Create the landing page for session management integration.

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

ghalliday committed:
Merge pull request #9371 from richardkchapman/javadoc

HPCC-16603 Extract Java Doc comments from attributes.

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge pull request #9359 from richardkchapman/coverity-leak3

HPCC-16428 Coverity: resource leak

Reviewed-By: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.4.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

jakesmith committed:
HPCC-16703 Ensure default keepStores match configmgr default

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

ghalliday committed:
Merge pull request #9023 from shamser/issue15926

HPCC-15926 Add BEST attribute to DEDUP

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.4.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9383 from jakesmith/hpcc-16703

HPCC-16703 Ensure default keepStores match configmgr default

Reviewed-By: Mark Kelly <mark.kelly@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
HPCC-16735 Refactoring to create MemberFunction objects

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

richardkchapman committed:
Merge branch 'candidate-6.4.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.4.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.4.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.4.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.4.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

Charles-Kaminski committed:
Fixed Spelling of Kaminski

Fixed Spelling of Kaminski

ghalliday committed:
HPCC-16735 Changes to comments following review

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

richardkchapman committed:
Merge pull request #9453 from Charles-Kaminski/patch-1

Fixed Spelling of Kaminski

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9448 from GordonSmith/HPCC-16804-FlatTheme

HPCC-16804 ECL Watch Flat Theme

Reviewed-By: Miguel Vazquez <miguel.vazquez@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

AttilaVamos committed:
HPCC-16630 Add expiry parameter to spray commands

Add 'expireDays' parameter to Spray, in Std.File ECl code and DfuPlus CLI.

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

richardkchapman committed:
Merge pull request #9401 from ghalliday/issue16735

HPCC-16735 Refactoring to create MemberFunction objects

Reviewed By: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.4.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.4.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

shamser committed:
HPCC-16171 String operation on varstring results in extra call to strlen

Signed-off-by: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>

ghalliday committed:
Merge pull request #9472 from shamser/issue16171

HPCC-16171 String operation on varstring results in extra call to strlen

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.4.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.4.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9267 from ghalliday/issue16517

HPCC-16517 Continue refactoring the roxie stats

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.4.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
HPCC-16886 Switch from max to min record size in field information

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
HPCC-16884 Fix core gathering stats from parallel loop graph

Regression introduced by HPCC-16517

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

richardkchapman committed:
Merge pull request #9449 from ghalliday/issue16805

HPCC-16805 Start refactoring field and type formats

Reviewed-By: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9496 from ghalliday/issue16884

HPCC-16884 Fix core gathering stats from parallel loop graph

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.4.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9495 from ghalliday/issue16886

HPCC-16886 Switch from max to min record size in field information

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
HPCC-16811 Changes following review

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

richardkchapman committed:
Merge pull request #9454 from ghalliday/issue16811

HPCC-16811 Provide functions to extract values from a typed field

Reviewed By: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

AttilaVamos committed:
Rework to use integer parameter

Add Regression Suite test case

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

xwang2713 committed:
HPCC-16796 Fix configurator coredump

Enable configurator and docs build integration
Change system/CMakeLists.txt to allow jlib only build for a component like configurator which
depends on it.

HPCCSmoketest commented:
Automated Smoketest
Sha: 5b05f611167358234b047d8d33ef29df202693c1
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 90.782 seconds |

Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unittest result:

| total | passed | failed | timeout | 
|---|---|---|---|
|86 |86 |0 |0 |

HPCC Stop: OK
HPCC Uninstall: OK

g-pan commented:
I have reviewed doc production paying close attention to the Using Configuration Manager doc. As the bulk of that is built from the configurator output. 
The doc looks to be correct. The files appear to reflect the appropriate attributes as found in the latest XSD files. I did not look at every page from every doc produced but from a good sized sampling focusing on the most complex doc. I approve.

richardkchapman committed:
Merge branch 'candidate-6.4.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-16901 ECLCC compile/link error if folding overflows

Also rationalize the divideByZero flag to apply to other functions such as
SQRT(-1), and provide a Std.Math module that contains some useful helper
functions related to infinity/nan.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman commented on configuration/configurator/BuildSet.cpp:
```diff
@@ -39,6 +40,9 @@ CBuildSetManager* CBuildSetManager::getInstance(const char* pBuildSetFile, const
 
         s_pBuildSetManager->m_buildSetPath.setf("%s%s%s", pBuildSetDirectory, pBuildSetDirectory[strlen(pBuildSetDirectory)-1] == '/' ? "" : "/", pBuildSetFile);
+
+        if (! (CFile(s_pBuildSetManager->m_buildSetPath.str())).exists())
```
it would be more normal and probably easier to read/follow the code if you just called checkFileExists() here

xwang2713 replied:
Thanks. I thought there should be a simple way to do it. :-)

richardkchapman commented on configuration/configurator/schemas/SchemaAttributes.cpp:
```diff
@@ -195,15 +195,17 @@ void CAttribute::populateEnvXPath(StringBuffer strXPath, unsigned int index)
 
-    const char *pChar = strrchr(strXPath.str(),'[');
-    assert(pChar != nullptr && strlen(pChar) >= 3);
+
```
Did you raise a ticket for it?

richardkchapman replied:
Not immediately obvious why it would core. But given the variable is not used (more commented out code) not sure why it was even there.

This code is a bit of a mess.

xwang2713 replied:
No, I didn't.  This change is from I run debug mode (assert enabled).
This is Gleb's new code. I think he didn't test for all xsd files. Some XPath strings do not have the pattern he assumed, i.e include '['. 
But actually later I found lots of asserts will throw exceptions.  So it cannot run in debug mode. 
I will remove this change.

richardkchapman commented on configuration/configurator/schemas/SchemaElement.cpp:
```diff
@@ -269,7 +269,9 @@ void CElement::getDocumentation(::StringBuffer &strDoc) const
     assert(strlen(this->getName()) > 0);
 
-    if (pGrandParentNode->getNodeType() == XSD_SCHEMA)
+    //if (pGrandParentNode->getNodeType() == XSD_SCHEMA)
```
Don't leave commented-out code in the sources.

richardkchapman reviewed (changes requested):
A few minor comments @xwang2713

ghalliday committed:
Merge pull request #9470 from AttilaVamos/HPCC-16630-new-feature-7.0.0

HPCC-16630 Add expiry parameter to spray commands

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #9506 from richardkchapman/fold_inf

HPCC-16901 ECLCC compile/link error if folding overflows

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
HPCC-16244 Update comment and freeBase following review

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

richardkchapman committed:
HPCC-16923 Added new funtions Std.Math.FMod and Std.Math.FMatch

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #9517 from richardkchapman/fmod

HPCC-16923 Added new funtions Std.Math.FMod and Std.Math.FMatch

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge pull request #9312 from ghalliday/issue16244

HPCC-16244 Implement delayed heaplet link counting

Reviewed-By: Mark Kelly <mark.kelly@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

AttilaVamos committed:
HPCC-16926 dbz3.ecl failed in OBT

Update keyfile based on the recent changes related to HPCC-16901

Force to run on roxie instead of (deprecated) hthor

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

richardkchapman committed:
Merge pull request #9521 from AttilaVamos/HPCC-16926-fix-7.0.0

HPCC-16926 dbz3.ecl failed in OBT

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-16900 Add multi-line string constants to ECL

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.4.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

xwang2713 committed:
HPCC-15414 clean lib2 for lib name changes on Mac OS
Before using install_name_tool 1) call otool to find out dependent lib name
2) find dependent lib name under lib2 directory
Also add fix for JIRA HPCC-16930: add dependent libraries for CBLAS support

ghalliday committed:
Merge pull request #9522 from richardkchapman/multiline

HPCC-16900 Add multi-line string constants to ECL

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.4.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9351 from jakesmith/hpcc-16637

HPCC-16637 Save old environment when update detected.

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman commented:
@xwang2713 This seems to have stalled

xwang2713 committed:
HPCC-16796 Use checkFileExists to check file
Remove the change for "assert(pChar..." but leave the comment.
"assert" is only enabled in debug mode. There are more than one or two places which cause
assert failures. The code may need a deeper look and re-design.
Remove commented-out code.

richardkchapman committed:
Merge branch 'candidate-6.4.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

HPCCSmoketest commented:
Automated Smoketest
Sha: 2e112bd64d9c17cbcf24235b1263a107a0c919f5
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 92.238 seconds |

Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unittest result:

| total | passed | failed | timeout | 
|---|---|---|---|
|86 |86 |0 |0 |

HPCC Stop: OK
HPCC Uninstall: OK

HPCCSmoketest commented:
Automated Smoketest
Sha: 3a34fcb89d297fab2839ad3996c57a9ab465c410
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 92.288 seconds |

Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unittest result:

| total | passed | failed | timeout | 
|---|---|---|---|
|86 |86 |0 |0 |

HPCC Stop: OK
HPCC Uninstall: OK

richardkchapman committed:
Merge pull request #9489 from miguelvazq/LoginPage

HPCC-14734 UI items login/logout in ECL Watch

Reviewed-By: Gordon Smith <gordon.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.4.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.4.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.4.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

GordonSmith committed:
HPCC-16820 Improved WU monitoring

Better "watch" releasing.

Fixes HPCC-16820

Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

xwang2713 committed:
HPCC-15414 Correct error message of CBLAS LIB dependency checking

richardkchapman committed:
HPCC-16995 PATTERN EscapeQuot := ''''; no longer compiles

Improve error message for unterminated multiline string constants.

Generate a warning if a (normal) string constant is followed by a ' - it
normally means someone was trying to use '' within a string constant to
indicate an escaped quote.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
HPCC-16931 Avoid loading dlls when extracting resources in esp

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

richardkchapman committed:
Merge branch 'candidate-6.4.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-17012 Update Pull request template

Updates as suggested

ghalliday committed:
Merge pull request #9575 from hpcc-systems/richardkchapman-patch-1

HPCC-17012 Update Pull request template

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
HPCC-16995 PATTERN EscapeQuot := ''''; no longer compiles

Add some test cases.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
HPCC-17026 Remove minor leaks from roxiemem unit tests

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
HPCC-17027 Initialise (unused) tm_wday and tm_mday in the tm struct

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

richardkchapman committed:
Merge branch 'candidate-6.4.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-17051 Coverity: Null dereference warning in cassandrawu.cpp

Add an assert to validate parameters passed in are consistent.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9535 from xwang2713/HPCC-15414_clean_lib2

HPCC-15414 clean lib2 for lib name changes on Mac OS

Reviewed-By: Michael Gardner <michael.gardner@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman commented:
@xwang2713 Can you rebase to squash the commits please

ghalliday committed:
Merge pull request #9561 from richardkchapman/tripleq2

HPCC-16995 PATTERN EscapeQuot := ''''; no longer compiles

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

HPCCSmoketest commented:
Automated Smoketest
Sha: f2962f8718d397e6803a051249df8f203eee3e6a
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 75 | 40.806 seconds |


Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |86 |86 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |


Regression test result:

| setup | hthor | thor | roxie |
| ----- | ----- | ----- | ---- |
| roxie_echo | Pass | Pass | Pass |
| setup | Pass | Pass | Pass |
| setup_fetch | Pass | Pass | Pass |
| setupdict | Pass | Pass | Pass |
| setupsearchindex | Pass | Pass | Pass |
| setupsq | Pass | Pass | Pass |
| setupwordindex | Pass | Pass | Pass |
| setupxml | Pass | Pass | Pass |
 

| test | hthor | thor | roxie |
| ----- | ----- | ----- | ---- |
| dbz3 | Excl. | Excl. | Pass |
| dedupwithbest | Pass | Pass | Pass |
| spray_expire_test | Pass | Excl. | Excl. |
| strings | Pass | Pass | Pass |
| teststdlibrary | Pass | Pass | Pass |
| zerodivide1 | Pass | Pass | Pass |
| zerodivide2 | Pass | Pass | Pass |
| zerodivide3 | Pass | Pass | Pass |
 
HPCC Stop: OK
HPCC Uninstall: OK

xwang2713 commented:
rebased with master. Should be candidate-6.4.0.  Recreate PR  #9591 to replace this
